### PR TITLE
Add faults [TFF-212]

### PR DESCRIPTION
**Pull request by Cafferoq**
### Summary

Server-side work for adding faults to items.

### Checklist

- [x] Have you added unit and functional tests as appropriate?
- [x] Did you update/add documentation for new methods or changed functionality?
- [x] Have you merged the target branch down into this one?

### How to Review

1. Check out the code.

### Links

- [TFF-212](https://msoese.atlassian.net/browse/TFF-212)

### CODE REVIEW

File: src/store/item-store.js [hunk 1]
```diff
@@ -10,6 +10,13 @@ let itemsByActionId = Object.create(null);
 
 class ItemStore extends Store {
 
+    addFault(address, fault){
```
Comment (gearedmonkey) [inline src/store/item-store.js]: store shouldn't be editing data. Move to a function outside the store

File: src/store/item-store.js [hunk 2]
```diff
@@ -110,4 +110,9 @@ store.registerHandler('DELETE_MODEL', data => {
     let itemsOfModel = store.getItems().filter(item => item.modelAddress === data.modelAddress);
     itemsOfModel.forEach(item => store.deleteItemByAddress(item.address));
 });
+
+store.registerHandler('ADD_ITEM_FAULT', data => {
+    store.getItemByAddress(data.itemAddress).isFaulty = true;
+    store.getItemByAddress(data.itemAddress).faultHistory.unshift(data.fault);
```
Comment (mattmahn) [inline src/store/item-store.js]: `unshift`, Javascript is weird

Comment (Cafferoq) [inline src/store/item-store.js]: Truth

Comment (thofmann) [inline src/store/item-store.js]: `unshift` is somewhat standard. It is weird, though, yes.

**Pull request merged**